python/pyarrow/table.pxi

```diff
@@ -467,7 +467,7 @@ cdef class ChunkedArray(_PandasConvertible):
         # pandas ExtensionDtype that implements conversion from pyarrow
         if hasattr(pandas_dtype, '__from_arrow__'):
             arr = pandas_dtype.__from_arrow__(self)
-            return pandas_api.series(arr)
+            return pandas_api.series(arr, name=self._name)
```

Comment:
Doesn't need to be done in this PR if we want to get this merged quickly for unblocking a release, but I think it would be good to add an explicit test for this (one that doesn't rely on some unrelated (and pandas-version dependent) test)

Reply:
Will add a separate issue so @raulcd can get the first RC going 👍